src/v/datalake/translation/scheduling.cc

```diff
@@ -78,22 +79,95 @@ class default_reservations_tracker : public reservations_tracker {
     }
 
     /*
-     * disk reservation overages are handled by notifying the scheduler, and
-     * then waiting for sufficient disk resources to become available.
+     * disk reservation overages are handled by requesting units from the global
+     * disk manager, and then waiting for sufficient disk resources to become
+     * available.
      */
     ss::future<reservation>
     reserve_disk(size_t, ss::abort_source& as) override {
+        const auto init_backoff = 100ms;
+        const auto max_backoff = 1000ms;
         const auto nr = _reservation_block_size;
```

Comment:
shouldn't we allocate a multiple of nr but to satisfy the requested size_t first param to the method? i don't see where this method returning less than requested is handled in the caller 🤔 https://github.com/redpanda-data/redpanda/pull/25766/commits/804b874b43aa6abb128df1226e39cfad2f028156#diff-b60aa20517b0824597059a188abfa745cf0af767dc421bf1504b91a1bae98da3R37

Reply:
The first parameter is present, pedantically, as it is technically a hint, but it isn't clear how to actually use it wisely. When it's "small" (e.g. 10s of KB) then it doesn't matter what its value is because we're going to hand back a larger reservation (e.g. single digit MBs).

On the other hand if it is large (e.g. above single digit MBs) then the cost of an extra round trip here is negligible and resources will be exhausted shortly anyway since the reservation is happening on a row-granularity.

Note that the memory reservation also ignores the size_t parameter (well, it technically it didn't even pass it in to begin with).

Reply:
was answered here https://github.com/redpanda-data/redpanda/pull/25766#issuecomment-2811075269

Reply:
Oh. I found the problem in my logic. reserve_bytes actually has a while loop to ensure it gets enough reservation. I thought the writer calls reserve_disk directly. All good!